### PR TITLE
September infra rollout - continued CoreCLR / Mono test cleanup (minimal)

### LINKED ISSUE #38919: August infra rollout: continued cleanup of runtime tests
Follow-up of:

https://github.com/dotnet/runtime/issues/37440

After the runtime test source code and project scripts have been moved under src/tests, I have the following proposals for further test cleanup:

1) Mop-up changes - moving the rest of src/coreclr/tests/src and ideally the entire src/coreclr/tests folder under src/tests;
2) Move build-test.cmd / build-test.sh from src/coreclr to src/tests; it might make sense to rename them to just build.cmd / build.sh in the process as the "test" part is implied by the folder and becomes superfluous in the script name;
3) Modify the logic for Crossgen1/Crossgen2 compilation of framework assemblies to r2rtest or, if that doesn't work, to MSBuild;
4) Start moving most of functional logic in the build-test scripts to the build.proj script with the ultimate goal of either keeping the two scripts as just thin command-line front-ends or deleting them altogether and triggering test build directly from the root build script. I don't think this is achievable for the August infra rollout in its entirety but it would be good to agree on the general process and start executing on it.

/cc: @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by trylek**
Similar to the previous occasion, I have identified this set as the minimum meaningful set of scripts to move without causing too much disruption apart from purely mechanical renames. This change moves the developer-facing test-related CoreCLR scripts under the <code>src/tests</code> folder in accordance with the previous move of the tests themselves:

<code>src/coreclr/build-test.cmd</code> to <code>src/tests/build.cmd</code>
<code>src/coreclr/build-test.sh</code> to <code>src/tests/build.sh</code>
<code>src/coreclr/tests/build.proj</code> to <code>src/tests/build.proj</code>
<code>src/coreclr/tests/runtest.cmd</code> to <code>src/tests/run.cmd</code>
<code>src/coreclr/tests/runtest.sh</code> to <code>src/tests/run.sh</code>
<code>src/coreclr/tests/src/runtest.proj</code> to <code>src/tests/run.proj</code>
<code>src/coreclr/tests/issues.targets</code> to <code>src/tests/issues.targets</code>

The rest of the change constitutes mere path adjustments to match the primary rename.

Move of the rest of the <code>src/coreclr/tests</code> folder under <code>src/tests</code> including unification of the various <code>props</code> and <code>targets</code> file that I originally proposed is extremely complicated and most of it doesn't affect normal developers' daily life at all so I believe that at least some of these cleanups can be made out-of-band, not waiting on the monthly infra roll-out cadence.

Fixes: https://github.com/dotnet/runtime/issues/38919

(I'll triage the remaining changes I originally proposed in the work item, I'll send out "normal" PR's for non-disruptive ones and I'll create a new follow-up September roll-out change for the rest.)

Thanks

Tomas

/cc: @dotnet/runtime-infrastructure

### CODE REVIEW

**Comment by trylek** [comment]
Hello everyone!

As I'm already three days past the originally planned roll-out date and there's been no progress on this review so far, I treat this as an indication that we're not inclined to risk destabilizing testing at this point and this change (and the related work item) should be postponed to the next monthly roll-out on September 7th. Please let me know if you think otherwise.

Thanks

Tomas

File: src/tests/build.proj [hunk 1]
```diff
@@ -3,12 +3,12 @@
   <Import Project="Directory.Build.props" />
 
   <ItemGroup>
-    <Project Include="src\dirs.proj" />
+    <Project Include="$(RepoRoot)\src\coreclr\tests\src\dirs.proj" />
```
Comment (ViktorHofer) [inline src/tests/build.proj:6]:
```suggestion
    <Project Include="$(CoreClrProjectRoot)tests\src\dirs.proj" />
```

File: src/tests/build.proj [hunk 2]
```diff
@@ -18,18 +18,18 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <RestoreProjects Include="$(MSBuildThisFileDirectory)scripts\scripts.csproj" />
-    <RestoreProjects Include="$(MSBuildThisFileDirectory)src\Common\test_dependencies_fs\test_dependencies.fsproj" />
-    <RestoreProjects Include="$(MSBuildThisFileDirectory)src\Common\test_dependencies\test_dependencies.csproj" />
+    <RestoreProjects Include="$(RepoRoot)src\coreclr\tests\scripts\scripts.csproj" />
+    <RestoreProjects Include="$(RepoRoot)src\coreclr\tests\src\Common\test_dependencies_fs\test_dependencies.fsproj" />
+    <RestoreProjects Include="$(RepoRoot)src\coreclr\tests\src\Common\test_dependencies\test_dependencies.csproj" />
     <RestoreProjects Include="$(RepoRoot)src\tests\Common\CoreCLRTestLibrary\CoreCLRTestLibrary.csproj" />
```
Comment (ViktorHofer) [inline src/tests/build.proj:24]:
```suggestion
    <RestoreProjects Include="$(MSBuildThisFileDirectory)Common\CoreCLRTestLibrary\CoreCLRTestLibrary.csproj" />
```

**Review by ViktorHofer — APPROVED** [review]

File: src/tests/build.proj (hunk 1 above)
Comment (trylek) [reply src/tests/build.proj]: Modified in 14th iteration.

File: src/tests/build.proj (hunk 2 above)
Comment (trylek) [reply src/tests/build.proj]: Changed in 14th commit.

**Comment by trylek** [comment]
Merging in as I believe that the CovariantReturn test failures are seen in other outerloop runs and unrelated to my change.

**Pull request merged**